src/views/sent_sms_list/SmsTable.js

```diff
@@ -12,69 +12,82 @@ import {
     TableFoot,
 } from '@dhis2/ui'
 import { PropTypes } from '@dhis2/prop-types'
+import Pagination from '../../pagination/Pagination'
 import SmsTableItem from './SmsTableItem'
-import Pagination from './Pagination'
-import s from './SmsTable.module.css'
+import styles from './SmsTable.module.css'
 
-const SmsTable = ({
-    smses,
-    toggleSelected,
-    toggleAll,
-    allSelected,
-    selected,
-    pager,
-}) => (
-    <Table>
-        <TableHead>
-            <TableRowHead>
-                <TableCellHead>
-                    <Checkbox
-                        disabled={smses.length === 0}
-                        onChange={toggleAll}
-                        checked={allSelected}
-                    />
-                </TableCellHead>
-                <TableCellHead>{i18n.t('Message')}</TableCellHead>
-                <TableCellHead>{i18n.t('Recipients')}</TableCellHead>
-                <TableCellHead>{i18n.t('Status')}</TableCellHead>
-                <TableCellHead>{i18n.t('Sent')}</TableCellHead>
-            </TableRowHead>
-        </TableHead>
-        <TableBody>
-            {smses.length === 0 ? (
+const SmsTable = ({ messages, pager, selectedIds, setSelectedIds }) => {
+    const selectedIdSet = new Set(selectedIds)
+    const allSelected =
+        messages.length > 0 && selectedIds.length === messages.length
+    const toggle = id => {
+        if (selectedIdSet.has(id)) {
+            selectedIdSet.delete(id)
+        } else {
+            selectedIdSet.add(id)
+        }
+
+        setSelectedIds(Array.from(selectedIdSet))
+    }
+    const toggleAll = () => {
+        if (allSelected) {
+            setSelectedIds([])
+        } else {
+            setSelectedIds(messages.map(({ id }) => id))
+        }
+    }
```

Comment:
It seems that these helper functions are now present for both the sent and the received sms tables. I think we could abstract those to a shared set of helpers for consistency, since they seem to fulfill the exact same role.

Reply:
Sounds good - I'll try to make a reusable `SmsTable` component that takes a child render function or just extract these into a group of selectors that can be imported.

Reply:
👍